src/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.hpp

```diff
@@ -0,0 +1,182 @@
+}  // namespace PUP
+/// \endcond
+
+namespace Elasticity::Solutions {
+/*!
+ * \brief The solution for a half-space mirror deformed by a laser beam.
+ *
+ * \details This solution is mapping thermal noise to an elasticity problem
+ * where a normally incident and axisymmetric laser beam with a Gaussian beam
+ * profile acts on the face of a semi-infinite mirror. Here we assume the face
+ * to be at \f$z = 0\f$ and the material to extend to \f$+\infty\f$ in the
+ * z-direction as well as for the mirror diameter to be comparetively large
+ * to the `beam width`. The mirror material is characterized by an isotropic
+ * homogeneous constitutive relation \f$Y^{ijkl}\f$ (see
+ * `Elasticity::ConstitutiveRelations::IsotropicHomogeneous`). In this scenario,
+ * the laser introduces a pressure in the form of a Gaussian distribution
+ *
+ * \f{align}
+ * T^{zz} &= \frac{e^{-\frac{r^2}{r_0^2}}}{\pi r_0^2} \\
+ * T^{xy} &= T^{yy} = 0 \text{.}
+ * \f}
+ *
+ * in the form of a Neumann boundary condition to the face of the mirror. We
+ * find that this stress in cylinder coordinates is produced by the displacement
+ * field
+ *
+ * \f{align}
+ * \xi_{r} &= \frac{1}{2 \mu} \int_0^{\infty} dk J_1(kr)e^{(-kz)}\left(1 -
+ * \frac{\lambda + 2\mu}{\lambda + \mu} + kz \right) \tilde{p}(k) \\
+ * \xi_{\phi} &= 0 \\
+ * \xi_{z} &=  \frac{1}{2 \mu} \int_0^{\infty} dk J_0(kr)e^{(-kz)}\left(1 +
+ * \frac{\mu}{\lambda + \mu} + kz \right) \tilde{p}(k)
+ * \f}
+ *
+ * and the strain
+ *
+ * \f{align}
+ * \Theta &= \frac{1}{2 \mu} \int_0^{\infty} dk
+ * J_0(kr) k e^{(-kz)}\left(\frac{-2\mu}{\lambda + \mu}\right) \tilde{p}(k) \\
+ * S_{rr} &= \Theta - S_{\phi\phi} - S_{zz} \\
+ * S_{\phi\phi} &= \frac{\xi_{r}}{r} \\
+ * S_{(rz)} &= -\frac{1}{2 \mu} \int_0^{\infty} dk J_1(kr) k e^{(-kz)}\left(kz
+ * \right) \tilde{p}(k) \\
+ * S_{zz} &= \frac{1}{2 \mu} \int_0^{\infty} dk
+ * J_0(kr) k e^{(-kz)}\left(-\frac{\mu}{\lambda + \mu} - kz \right) \tilde{p}(k)
+ * \f}
+ *
+ * (Eqs. 11 and 12 in \cite Lovelace2007tn), where \f$ \tilde{p}(k)=
+ * \frac{1}{2\pi}e^{(\frac{kr}{2})^2}\f$ is the Hankel-Transform of the lasers
+ * intensity profile and \f$ \Theta = \mathrm{Tr}(S)\f$ the materials expansion.
+ *
+ */
+class HalfSpaceMirror {
+ public:
+  static constexpr size_t volume_dim = 3;
+
+  using constitutive_relation_type =
+      Elasticity::ConstitutiveRelations::IsotropicHomogeneous<3>;
+
+  struct BeamWidth {
+    using type = double;
+    static constexpr OptionString help{"The lasers beam width"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  struct Material {
+    using type = constitutive_relation_type;
+    static constexpr OptionString help{"The material properties of the beam"};
+  };
+
+  struct IntegrationIntervals {
+    using type = size_t;
+    static constexpr OptionString help{
+        "Workspace size for numerical integrals"};
+    static type lower_bound() noexcept { return 1; }
+    static type default_value() noexcept { return 350; }
```

Comment:
Where do these limits come from?

Reply:
@tomwlodarczyk please add some information for the user to the help string. I suggest something like this: "Workspace size for numerical integrals. Increase if integration fails at distances much larger than the beam width. The default value should accommodate distances of up to ~100 beam widths." (Feel free to suggest further changes to this text if it's still unclear to you @geoffrey4444)